client/src/app/services/tasks.service.ts

```diff
@@ -1,19 +1,31 @@
 import { Injectable } from "@angular/core";
 import { Task } from "../models/task";
 import { HttpClient } from "@angular/common/http";
-import { firstValueFrom, take } from "rxjs";
+import { first, firstValueFrom, take } from "rxjs";
+import { TasksComponent } from "../tasks/tasks.component";
 
 @Injectable({ providedIn: 'root'})
 export class TasksService {
   tasks: Task[] = [];
 
   constructor(private http: HttpClient) {}
-
-  async getAll() {
+  
+  async getAll() { // This function gets all the tasks from the server 
     this.tasks = await firstValueFrom(this.http.get<Task[]>('/api/tasks/').pipe(take(1)));
   }
 
-  create(name: string) {
+  create(name: string) { // This function creates the new task
     return firstValueFrom(this.http.post<void>('/api/tasks/', { name }).pipe(take(1)));
   }
+
+  async remove(id: number) { //This task deletes a task at a specific id
+    await firstValueFrom(this.http.delete<void>(`/api/tasks/${ id }`, ))
+    await this.tasks.pop(); //deletes the task completely
```

Comment:
Let's not use `pop` because that's always going to remove the last task. Instead you could just 

`await this.getAll()`